### DISCUSSION TITLE
Open file in a new tab

### DISCUSSION THREAD

**Discussion by tdwong** [coteditor/CotEditor#1343] [Q&A]
With an existing document opened, how can I open another document into a new tab (of the same window)?
The "File > Open" option always open the second document in a new window.

**Comment by 1024jp** [comment]
There is currently no way to open an existing document in a new tab as necessary, but you can change the entire behavior to open documents always in tabs by setting it in Preferences > Window > Prefer Tab. Is this fulfill your purpose?

<img width="605" alt="Screenshot 2022-08-11 at 1 28 25" src="https://user-images.githubusercontent.com/1165044/183963507-44ab5e22-332e-4ffd-a72a-614df8c67c62.png">

**Reply by gitblanc** [reply]
it happens the same to me :/, would be a nice idea to improve

**Comment by johnmapeson** [comment]
@1024jp Maybe at least allow user to drag a file from Finder to CotEditor tab bar? And the same way, to allow drag one CotEditor window to the tab bar of the another its window? This is how it works in Xcode and many tabbed text editors.

**Reply by 1024jp** [reply]
I know it would be nice, but Apple doesn't provide any APIs for that. ref.: #784

**Comment by itsamemarkus** [comment]
Hi, I'm very new to CotEditor and I was testing a few text editors for some use cases I had with Notepad++ on Windows before. I'm glad that I found this info. In my opinion new tabs should be the default behavior. Can someone explain to me which system setting this refers to?

![Screenshot 2024-07-14 at 19 15 33@2x](https://github.com/user-attachments/assets/6f1f14fd-5d6d-435e-9bd4-376a0f0a8237)

**Reply by johnmapeson** [reply]
> Can someone explain to me which system setting this refers to?

Open the "System Settings" app, then click "Desktop & Dock" and scroll to the "Windows" section. The setting itself is called "Prefer tabs when opening documents".

The "System Settings" app has a search field, so instead of to click "Desktop & Dock" and then scroll to the "Windows" section, you can simply type "tabs" in that search field, and that's it.

I'm just a casual user like you and not a CotEditor developer. But keep in mind that general-purpose questions like this one should be asked on general-purpose macOS-related websites, such as Ask Different or Super User, and not here.

**Comment by itsamemarkus** [comment]
Hi @jsx97 thanks for the quick reply. The reason why I posted here was more for making a statement that tabs should be the default from the beginning regardless of system settings, because I think CotEditor could be an awesome replacement for users coming from Windows that are searching for a Notepad++ alternative, which is often used for dumping notes in new tabs. I often read about users searching something like that when they get try macOS TextEdit and get the `Do you want to keep this new document “Untitled.txt”?` prompt. (although CotEditor can do much more than that of course)

![Screenshot 2024-07-14 at 21 37 09](https://github.com/user-attachments/assets/455c015c-0e39-433f-8d09-f1dc9954b5a5)

So I sneaked in the macOS newbie question, because I'm new to macOS. :) Thanks for your answer
